11_notify/n0npax/pkg/puppy/store/syncStore.go

```diff
@@ -0,0 +1,69 @@
+package store
+
+import (
+	"sync"
+
+	puppy "github.com/anz-bank/go-course/11_notify/n0npax/pkg/puppy"
+)
+
+//SyncStore sync.Map based type for storing puppies data
+type SyncStore struct {
+	sync.Map
+	sync.Mutex
+	nextID int
+}
+
+// NewSyncStore creates new storer for SyncMap
+func NewSyncStore() *SyncStore {
+	return &SyncStore{}
+}
+
+// CreatePuppy creates puppy
+func (s *SyncStore) CreatePuppy(p *puppy.Puppy) (int, error) {
+	if p.Value < 0 {
+		return -1, puppy.Errorf(puppy.ErrCodeInvalidInput, "puppy value have to be positive number")
+	}
+	s.Lock()
+	defer s.Unlock()
+	p.ID = s.nextID
+	s.nextID++
+	s.Store(p.ID, *p)
+	return p.ID, nil
+}
+
+// ReadPuppy reads puppy from backend
+func (s *SyncStore) ReadPuppy(id int) (*puppy.Puppy, error) {
+	if puppyData, ok := s.Load(id); ok {
+		p := puppyData.(puppy.Puppy)
+		return &p, nil
+	}
+	return nil, puppy.Errorf(puppy.ErrCodeNotFound, "puppy with ID (%v) not found", id)
+}
+
+// UpdatePuppy updates puppy
+func (s *SyncStore) UpdatePuppy(id int, p *puppy.Puppy) error {
+	if p.Value < 0 {
+		return puppy.Errorf(puppy.ErrCodeInvalidInput, "puppy value have to be positive number")
+	}
+	if id != p.ID {
+		return puppy.Errorf(puppy.ErrCodeInvalidInput, "ID is corrupted. Please ensure object ID matched provided ID")
+	}
+	s.Lock()
+	defer s.Unlock()
+	if _, ok := s.Load(id); !ok {
+		return puppy.Errorf(puppy.ErrCodeNotFound, "puppy with ID (%v) not found", id)
+	}
+	s.Store(id, *p)
+	return nil
+}
+
+// DeletePuppy deletes puppy
+func (s *SyncStore) DeletePuppy(id int) (bool, error) {
+	s.Lock()
+	defer s.Unlock()
+	if _, ok := s.Load(id); !ok {
+		return false, puppy.Errorf(puppy.ErrCodeNotFound, "puppy with ID (%v) not found", id)
+	}
+	s.Delete(id)
+	return true, nil
```

Comment:
i think it's nicer just to return an error (no need for bool as well).

Reply:
yes in did. Shame on me :(. Will be fixed soon. Thanks!

Reply:
I've no idea why I did this. I even provided test for this :man_facepalming: Anyway, fixed